pkg/operator/target_config_reconciler.go

```diff
@@ -146,38 +147,51 @@ func (c TargetConfigReconciler) sync() error {
 					Status: operatorv1.ConditionTrue,
 				}))
 		}
-		return err
+	} else {
+		specAnnotations["configmaps/cluster"] = configMap.ObjectMeta.ResourceVersion
```

Comment:
Worth double checking https://github.com/openshift/cluster-openshift-controller-manager-operator/commit/5e87d418fa9e170ee615109b9875b92427d7cd16 whether all objects are non-nil. The same for SSO.